src/Microsoft.ML.StandardTrainers/Standard/MulticlassClassification/MulticlassNaiveBayesTrainer.cs

```diff
@@ -339,22 +339,23 @@ private protected override void SaveCore(ModelSaveContext ctx)
 
             // *** Binary format ***
             // int: _labelCount
-            // int[_labelCount]: _labelHistogram
+            // long[_labelCount]: _labelHistogram
             // int: _featureCount
-            // int[_labelCount][_featureCount]: _featureHistogram
+            // long[_labelCount][_featureCount]: _featureHistogram
             // int[_labelCount]: _absentFeaturesLogProb
-            ctx.Writer.WriteIntArray(_labelHistogram.AsSpan(0, _labelCount));
+            ctx.Writer.Write(_labelCount);
+            ctx.Writer.WriteLongStream(_labelHistogram);
             ctx.Writer.Write(_featureCount);
             for (int i = 0; i < _labelCount; i += 1)
             {
                 if (_labelHistogram[i] > 0)
-                    ctx.Writer.WriteIntsNoCount(_featureHistogram[i].AsSpan(0, _featureCount));
+                    ctx.Writer.WriteLongStream(_featureHistogram[i]);
```

Comment:
Sames as above.

`.AsSpan()` is not implemented for converting from 'System.Span<long>' to 'System.Collections.Generic.IEnumerable<long>. The only other way around this is using `Array.Copy()`, but that introduces needless array copying. I also don't see a case where not all of `_labelHistogram` is not serialized. As such, `_labelHistogram` is always serialized whole here.